src/repository/MongoRepository.ts

```diff
@@ -205,7 +205,7 @@ export class MongoRepository<
     aggregate<R = any>(
         pipeline: ObjectLiteral[],
         options?: AggregateOptions,
-    ): AggregationCursor<Entity> {
+    ): AggregationCursor<R> {
```

Comment:
@jharmelink thanks for your effort on this.
I'm closing this for now because it's stale, feel free to create a new PR with your improvements. 💪